public/app/features/alerting/partials/alert_tab.html

```diff
@@ -30,6 +31,7 @@ <h4 class="section-heading">Rule</h4>
           ng-model="ctrl.alert.for"
           spellcheck="false"
           placeholder="5m"
+          ng-pattern="^\d+([dhms])$"
```

Comment:
Need to allow an empty value and only zero (0). Zero is stored in database and rendered in UI if For has been saved with an empty value.
![image](https://user-images.githubusercontent.com/1668778/113167350-efe29480-9243-11eb-9b13-47593e7cbe77.png)

```suggestion
          ng-pattern="/^\d+([dhms])$/"
```